python_modules/dagster/dagster/_core/executor/base.py

```diff
@@ -42,6 +42,37 @@ def retries(self) -> RetryMode:
         Returns: RetryMode
         """
 
+    def get_step_event_or_retry_event(
```

Comment:
not blocking, but i wonder if there is an opportunity to share more code with get_failure_or_retry_event_after_crash? seems like the only difference is what event gets returned if it shoudn't retry (I guess its tricky because just creating the step failure event also creates it, right?)

Reply:
yeah, I originally had it share more code, but just creating events will cause them to be emitted in the step delegating executors (that one took longer than it should for me to debug).